src/include/sof/audio/codec_adapter/codec/cadence.h

```diff
@@ -36,13 +36,24 @@ struct cadence_api {
 	xa_codec_func_t *api;
 };
 
+enum default_stream_params_ids {
+	CADENCE_SAMPLE_RATE_ID = 0x01,
+	CADENCE_SAMPLE_WIDTH_ID = 0x02,
+	CADENCE_CHANNELS_COUNT_ID = 0x04
+}
+
 struct cadence_codec_data {
 	char name[LIB_NAME_MAX_LEN];
 	void *self;
 	xa_codec_func_t *api;
 	void *mem_tabs;
+	uint32_t sample_width_id;
+	uint32_t sample_rate_id;
+	uint32_t channels_id;
 };
 
+
+
 /*****************************************************************************/
```

Comment:
remove the new lines from here. Also, add the following changes:

```
--- a/src/audio/codec_adapter/CMakeLists.txt
+++ b/src/audio/codec_adapter/CMakeLists.txt
@@ -4,6 +4,7 @@ add_local_sources(sof codec_adapter.c codec/generic.c)
 
 if(CONFIG_CADENCE_CODEC)
 add_local_sources(sof codec/cadence.c)
+target_include_directories(sof PUBLIC ${CMAKE_CURRENT_LIST_DIR}/../../include/sof/audio/cadence)
```

 ```
--- a/src/include/sof/audio/codec_adapter/codec/cadence.h
+++ b/src/include/sof/audio/codec_adapter/codec/cadence.h
@@ -13,6 +13,9 @@
 #include <sof/audio/cadence/xa_error_standards.h>
 #include <sof/audio/cadence/xa_memory_standards.h>
 
+#include <sof/audio/cadence/mp3_dec/xa_mp3_dec_api.h>
+#include <sof/audio/cadence/aac_dec/xa_aac_dec_api.h>
```

I will add a PR with these changes later if you want.